types/foundry/common/abstract/backend.d.ts

```diff
@@ -0,0 +1,238 @@
+/**
+ * An interface shared by both the client and server-side which defines how creation, update, and deletion operations are transacted.
+ */
+export default abstract class DatabaseBackend<TDocument extends foundry.abstract.Document> {
+    /* -------------------------------------------- */
+    /*  Get Operations                              */
+    /* -------------------------------------------- */
+
+    /**
+     * Retrieve Documents based on provided query parameters
+     * @param documentClass   The Document definition
+     * @param context         Context for the requested operation
+     * @param [user]          The requesting User
+     * @returns               The created Document instances
+     */
+    get(
+        documentClass: ConstructorOf<TDocument>,
+        context: Partial<DatabaseBackendGetContext>,
+        user?: User
+    ): Promise<TDocument[]>;
+
+    /**
+     * Validate the arguments passed to the get operation
+     * @param context                The requested operation
+     * @param [context.query={}]     A document search query to execute
+     * @param [context.options={}]   Operation options
+     * @param [context.pack]         A Compendium pack identifier
+     */
+    private _getArgs(context?: Partial<DatabaseBackendGetContext>): Promise<DatabaseBackendGetContext & WithParent>;
+
+    /** Get primary Document instances */
+    protected abstract _getDocuments(
+        documentClass: ConstructorOf<TDocument>,
+        request: DatabaseBackendGetContext,
+        user: User
+    ): Promise<(DeepPartial<TDocument["_source"][]> & CompendiumIndexData) | TDocument[]>;
+
+    /**
+     * Get the parent Document (if any) associated with a request context.
+     * @param context    The requested operation
+     * @return           The parent Document, or null
+     */
+    private _getParent(
```

Comment:
?

Reply:
```js
  /**
   * Get the parent Document (if any) associated with a request context.
   * @param {object} context                The requested operation
   * @return {Promise<Document|null>}       The parent Document, or null
   * @private
   */
  async _getParent(context) {
    if ( !context.parent ) return null;
    if ( !(context.parent instanceof Document) ) {
      throw new Error("A parent Document provided to the database operation must be a Document instance");
    }
    return context.parent;
  }
  ```

Reply:
If it's truly private, it doesn't need to be there since it's not part of the actual interface.

Reply:
Ok, I wasn't sure what to do with the `@private` methods. I'll remove them when I'm home.

Reply:
They are gone